source/FilesServer/app/endpoints.py

```diff
@@ -23,3 +27,14 @@ async def download(file_name: str) -> RedirectResponse:
 @router.post("/upload", status_code=status.HTTP_201_CREATED)
 async def upload(file: UploadFile) -> None:
     await service.s3_upload(file)
+
+
+@router.get("/metadata", status_code=status.HTTP_200_OK)
+async def metadata(file_name: str) -> FileMetadata:
+    if not file_name:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="No file name provided"
+        )
+    return await service.s3_get_metadata(
```

Comment:
this function might return a None, you need to handle this

Reply:
done